server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/ModularizeJmapDraftAuthenticationStrategyTest.java

```diff
@@ -0,0 +1,260 @@
+import static org.hamcrest.Matchers.both;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.isA;
+import static org.hamcrest.Matchers.notNullValue;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import org.apache.james.GuiceJamesServer;
+import org.apache.james.core.Username;
+import org.apache.james.jmap.draft.JmapGuiceProbe;
+import org.apache.james.jmap.http.AuthenticationStrategy;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.utils.DataProbeImpl;
+import org.hamcrest.CoreMatchers;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.restassured.RestAssured;
+
+public abstract class ModularizeJmapDraftAuthenticationStrategyTest {
+    public static String DOMAIN = "domain.tld";
+    public static Username BOB = Username.of("bob" + "@" + DOMAIN);
+    public static String BOB_PASSWORD = "123456";
+
+    private AllowAuthenticationStrategy allowAuthenticationStrategy;
+    private DenyAuthenticationStrategy denyAuthenticationStrategy;
+    private GuiceJamesServer jmapServer;
+    private AccessToken bobAccessToken;
+
+    protected abstract GuiceJamesServer createJmapServerWithAllowAuthenticationStrategy(AllowAuthenticationStrategy authenticationStrategy) throws IOException;
+
+    protected abstract GuiceJamesServer createJmapServerWithDenyAuthenticationStrategy(DenyAuthenticationStrategy authenticationStrategy) throws IOException;
+
+    protected abstract GuiceJamesServer createJmapServerWithDefaultAuthenticationStrategies() throws IOException;
+
+    protected abstract MailboxManager initMailboxManager();
+
+    @Before
+    public void setup() {
+        MailboxManager mailboxManager = initMailboxManager();
+        allowAuthenticationStrategy = new AllowAuthenticationStrategy(mailboxManager);
+        denyAuthenticationStrategy = new DenyAuthenticationStrategy();
+    }
+
+    public void setupJamesServerWithCustomAuthenticationStrategy(Optional<AuthenticationStrategy> authenticationStrategy) throws Throwable {
+        jmapServer = authenticationStrategy.map(strategy -> {
+                if (strategy instanceof AllowAuthenticationStrategy) {
+                    try {
+                        return createJmapServerWithAllowAuthenticationStrategy((AllowAuthenticationStrategy) strategy);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                } else if (strategy instanceof DenyAuthenticationStrategy) {
+                    try {
+                        return createJmapServerWithDenyAuthenticationStrategy((DenyAuthenticationStrategy) strategy);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                }
+                return null;
+            }).orElse(createJmapServerWithDefaultAuthenticationStrategies());
```

Comment:
We should be able to just pass the list of authentication strategies we want for Draft and RFC-8621.

From there, we could craft the JMAPDraft configuration POJO and the JMAPRFC8621 configuration POJO  with those strategies and inject these strategies using a guice module override. 

We would be closer to reality!